web_poet/requests.py

```diff
@@ -0,0 +1,184 @@
+"""This module has a full support for :mod:`asyncio` that enables developers to
+perform asynchronous additional requests inside of Page Objects.
+
+Note that the implementation to fully execute any :class:`~.Request` is not
+handled in this module. With that, the framework using **web-poet** must supply
+the implementation.
+
+You can read more about this in the :ref:`advanced-downloader-impl` documentation.
+"""
+
+import asyncio
+import logging
+from contextvars import ContextVar
+from typing import Optional, List, Dict, Union, Callable, AnyStr, TypeVar, Type
+
+import attrs
+from multidict import CIMultiDict
+
+from web_poet.page_inputs import HttpResponse
+from web_poet.exceptions import RequestBackendError
+
+logger = logging.getLogger(__name__)
+
+T_body = TypeVar("T_body", bound="HttpRequestBody")
+
+
+# Frameworks that wants to support additional requests in ``web-poet`` should
+# set the appropriate implementation for requesting data.
+request_backend_var: ContextVar = ContextVar("request_backend")
+
+
+class HttpRequestBody(bytes):
+    """A container for holding the raw HTTP request body in bytes format."""
+
+    pass
+
+
+class HttpRequestHeaders(CIMultiDict):
```

Comment:
Why is the implementation different from HttpResponseHeaders? I haven't checked it, but wouldn't it make sense to have HttpHeaders class, and then have request/response header classes inherit from it?

Reply:
Good point, I was thinking of the same thing. I've addressed this in https://github.com/scrapinghub/web-poet/pull/22/commits/65badca5d6746e0d96ee09045b33e0bbf4556c87.